backend/tarsy/agents/iteration_controllers/base_controller.py

```
@@ -317,12 +392,16 @@ async def run_iteration():
                     # Extract native tools override from context (if specified)
                     native_tools_override = self._get_native_tools_override(context)
                     
+                    # Get parallel execution metadata for streaming
+                    parallel_metadata = context.agent.get_parallel_execution_metadata()
+                    
                     conversation_result = await self.llm_manager.generate_response(
                         conversation=conversation,
```

Ruff (0.14.8)

Function definition does not bind loop variable `conversation`

(B023)